woodwardmatt/paid-memberships-pro
# Removing deprecated stripe-php function calls

### All Submissions:






### Changes proposed in this Pull Request:



This change removed calls to a deprecated function in the Stripe-PHP library.

The deprecated call can be seen here:  https://github.com/strangerstudios/paid-memberships-pro/blob/00e7e8130ad667808c161f0c3964ae732798a3df/classes/gateways/class.pmprogateway_stripe.php#L1940

This function (i.e. Stripe\Customer::invoices()) was deprecated in the Stripe PHP Library in this commit: https://github.com/stripe/stripe-php/commit/8a046cfd119e50a048645f8d4249490a566379bc#diff-814438c52fc09aaad9ad47d5fa40351c

Having queries this with the Stripe support team, the recommended approach is now to use the list function on the Invoice object with the appropriate customer id and status (of open). This approach is documented at:  https://stripe.com/docs/api/invoices/list?lang=php

### How to test the changes in this Pull Request:

1. Create a new member & membership
2. Whilst logged in as that member, choose to cancel the membership
3. See error "There has been a critical error on your website."

### Other information:




### Changelog entry

Remove calls to deprecated stripe-php function Stripe\Customer::invoices() used during the cancellation of a membership.
